spec.md

```diff
@@ -2127,6 +2127,20 @@ volumes:
 An entry under the top-level `volumes` key can be empty, in which case it uses the platform's default configuration for
 creating a volume. Optionally, you can configure it with the following keys:
 
+### artifact
+
+Specify reference to an [OCI artifact](https://github.com/opencontainers/artifacts) used to populate volume when created.
+When set, a Compose implementation MUST pull the OCI artifact from reference and populate newly created volume with content.
```

Comment:
Should we use the same behavior as `image`? Default pull policy of "missing" with option to specify via long syntax?

https://github.com/compose-spec/compose-spec/blob/1eb16a20675324b35258b74b8375b48399893eab/spec.md?plain=1#L1556-L1568

```yaml
volumes:
  example:
    oci:
      artifact: "registry-1.docker.io/docker/demo:0.0.1"
      pull_policy: "always"
```

Reply:
A challenge as we introduce `pull_policy` here is that Docker engine has no local cache for OCI artifacts, so Compose will need to manage his own, or just ignore this field. Maybe better keep it for a future evolution of the spec as alternatives get developed ?